edi_oca/tests/test_security.py

```diff
@@ -0,0 +1,223 @@
+        self.assertTrue(exchange_record.with_user(self.user).unlink())
+
+    @mute_logger("odoo.addons.base.models.ir_rule")
+    def test_rule_no_unlink(self):
+        exchange_record = self.create_record()
+        self.user.write({"groups_id": [(4, self.group.id)]})
+        self.consumer_record.name = "no_rule"
+        with self.assertRaisesRegex(AccessError, "doesn't have 'write' access to"):
+            exchange_record.with_user(self.user).unlink()
+
+    def test_no_group_no_search(self):
+        exchange_record = self.create_record()
+        self.assertEqual(
+            0,
+            self.env["edi.exchange.record"]
+            .with_user(self.user)
+            .search_count([("id", "=", exchange_record.id)]),
+        )
+
+    def test_group_search(self):
+        exchange_record = self.create_record()
+        self.user.write({"groups_id": [(4, self.group.id)]})
+        self.assertEqual(
+            1,
+            self.env["edi.exchange.record"]
+            .with_user(self.user)
+            .search_count([("id", "=", exchange_record.id)]),
+        )
+
+    def test_rule_no_search(self):
+        exchange_record = self.create_record()
+        self.user.write({"groups_id": [(4, self.group.id)]})
+        self.consumer_record.name = "no_rule"
+        self.assertEqual(
+            0,
+            self.env["edi.exchange.record"]
+            .with_user(self.user)
+            .search_count([("id", "=", exchange_record.id)]),
+        )
+
+    def test_search_no_record(self):
+        # Consumer record no longer exists:
+        #  exchange_record is hidden in search
+        exchange_record = self.create_record()
+        exchange_record.res_id = -1
+        self.user.write({"groups_id": [(4, self.group.id)]})
+        logger_name = "odoo.addons.edi_oca.models.edi_exchange_record"
+        expected_msg = (
+            f"WARNING:{logger_name}:"
+            f"Deleted record {exchange_record.model},{exchange_record.res_id} "
+            f"is referenced by edi.exchange.record [{exchange_record.id}]"
+        )
+        with self.assertLogs(logger_name, "WARNING") as watcher:
+            self.assertEqual(
+                0,
+                self.env["edi.exchange.record"]
+                .with_user(self.user)
+                .search_count([("id", "=", exchange_record.id)]),
+            )
+            self.assertEqual(watcher.output, [expected_msg])
+
+    def test_search_no_record_admin(self):
+        # Consumer record no longer exists:
+        #  user with group "Settings" has access
+        exchange_record = self.create_record()
+        exchange_record.res_id = -1
+        admin_group = self.env.ref("base.group_system")
+        self.user.write({"groups_id": [(4, self.group.id), (4, admin_group.id)]})
+        self.assertEqual(
+            1,
+            self.env["edi.exchange.record"]
+            .with_user(self.user)
+            .search_count([("id", "=", exchange_record.id)]),
+        )
```

Comment:
Changing this will solve the problem you are having. :wink:
```suggestion
        logger_name = "odoo.addons.edi_oca.models.edi_exchange_record"
        with self.assertLogs(logger_name, "WARNING"):
            self.assertEqual(
                1,
                self.env["edi.exchange.record"]
                .with_user(self.user)
                .search_count([("id", "=", exchange_record.id)]),
            )
```

Reply:
Thank you @thienvh332 🙌🏼